### ISSUE TITLE
cannot find function `get_region` in module `os`

### ISSUE THREAD

**Issue by alterstep** [fastly/Viceroy#4]
```
error[E0425]: cannot find function `get_region` in module `os`
   --> /home/jochen/.cargo/registry/src/github.com-1ecc6299db9ec823/region-2.2.0/src/lib.rs:133:7
    |
133 |   os::get_region(page::floor(address as usize) as *const u8)
    |       ^^^^^^^^^^ not found in `os`

   Compiling jobserver v0.1.22
error: aborting due to previous error
```

**Comment by cratelyn** [comment]
Hi! Thanks for filing an issue @alterstep. Would you mind adding a bit more information about what was happening when you ran into this error? What platform you're using might be particularly important information here. :slightly_smiling_face:

**Comment by alterstep** [comment]
* OS is OpenBSD 6.9 (GENERIC.MP amd64)
* rust version is 1.53.0

**Comment by cratelyn** [comment]
Alright, I found some more information about this! Thanks for the additional information. Unfortunately, (_and I say this as a BSD fan_) **I don't believe we can yet commit to support for OpenBSD.**

`wasmtime`, which is used internally by this tool, is still in the process of introducing OpenBSD support. bytecodealliance/wasmtime#2980 adds OpenBSD support, but that is in turn blocked on other upstream work. One noteworthy detail from that issue is that `rustup` doesn't carry toolchain builds for OpenBSD, so building/testing in CI consequently isn't possible.

Upstream BSD support appears to be a low priority at the moment, and I am afraid that's not likely to change soon.

For now, I believe building/running Viceroy inside of a VM or Docker container would be the best solution. I'm sorry that we can't provide a better answer @alterstep, and thanks again for filing this issue.

**Comment by cratelyn** [comment]
I'm going to close this for now. #35 introduced CI coverage for the other platforms that we guarantee support for, but we'll need to wait on upstream/ecosystem work to occur before we can revisit support for BSD variants ourselves.
